google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkServiceType.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+/** Supported {@link CloudSdkService} types. */
+public enum CloudSdkServiceType {
```

Comment:
something intuitively feels like this whole strategy could be more more OO. But I guess since we are persisting strings and we then need some way to return the appropriate Service from the string there isn't really a better approach.

Reply:
Agree, it feels more of the settings things. We can probably use class name / generic settings object per SDK interface, but then again, have to specify them directly from Cloud SDK settings UI. I feel it would be easy to generalize that if we need to (i.e. more implementations of SDK service), but for now it doesn't feel we'll have any more anytime soon.